cellular_configs/v1alpha1/plmn_types.go

```diff
@@ -0,0 +1,97 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PLMNStatus defines the observed state of PLMN
+type PLMNStatus struct {
+}
+
+// +kubebuilder:object:root=true
+type PLMN struct {
+	metav1.TypeMeta   `json:",inline" yaml:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	Spec   PLMNSpec   `json:"spec,omitempty" yaml:"spec,omitempty"`
+	Status PLMNStatus `json:"status,omitempty" yaml:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// PLMNList contains a list of PLMNs
+type PLMNList struct {
+	metav1.TypeMeta `json:",inline" yaml:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+	Items           []PLMN `json:"items" yaml:"items"`
+}
+
+// PLMNSpec defines the characteristics of a deployment of a network function
+type PLMNSpec struct {
+	// PLMNInfo defines the list of PLMN to configure for core NFs
+	PLMNInfo []PLMNInfo `json:"PLMNInfo" yaml:"PLMNInfo"`
+}
+
+// PLMNInfo defines the structure of PLMN
+type PLMNInfo struct {
+	// PLMNID defines the PLMN Identifier
+	PLMNID PLMNID `json:"plmnID" yaml:"plmnID"`
+	// tac defines the tracking area code
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=16777215
+	TAC uint32 `json:"tac" yaml:"tac"`
+	// NSSAI defines the Network Slice Selection Assistance Information
+	NSSAI []NSSAI `json:"nssai" yaml:"nssai"`
+}
+
+// PLMNID defines the Public Land Mobile Network Identifier
+type PLMNID struct {
+	// mcc defines the mobile country code
+	// +kubebuilder:validation:Pattern=`[02-79][0-9][0-9]`
+	MCC string `json:"mcc" yaml:"mcc"`
+	// mnc defines the mobile network code
+	// +kubebuilder:validation:Pattern=`[0-9][0-9][0-9]|[0-9][0-9]`
+	MNC string `json:"mnc" yaml:"mnc"`
+}
+
+// NSSAI defines the Network Slice Selection Assistance Information
+type NSSAI struct {
+	// Sst defines Slice/Service Type
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=255
+	SST int `json:"sst" yaml:"sst"`
+	// Sd defines Service Differentiator
+	// +optional
+	// +kubebuilder:validation:Pattern=`^[A-Fa-f0-9]{6}$`
+	SD *string `json:"sd,omitempty" yaml:"sd,omitempty"`
+	// dnnInfo defines the Data Network Names information
+	DNNInfo DNNInfo `json:"dnnInfo" yaml:"dnnInfo"`
+}
+
+// NOTE: For R2 the DNN.name here should match with NF deployment DNN name. This is a hack for R2
+// DNN defines the Data Network Names Information
+type DNNInfo struct {
+	// name defines DNN
+	Name string `json:"name" yaml:"name"`
+	// SessionType defines session type
+	// +kubebuilder:validation:Enum=ipv4;ipv6;ipv4v6;unstructured;ethernet
+	SessionType string `json:"sessionType" yaml:"sessionType"`
+	// DNS defines Domain Name Server
+	// +optional
+	DNS *string `json:"dns" yaml:"dns"`
```

Comment:
I am still not clear on what this is. Is it the resolver address the CPE should use? Is it an IP address? If so, maybe "NameServerIPAddress"? Can there be more than one? Then it should be plural.

Reply:
@johnbelamaric It is an IP Address and what you understand is correct (NameServerIPAddress). I used the term DNS because most of the configuration files (Free5GC, OAI, Open5GS Core Network) use it as `dns`.

I have attached the screenshot of the field from the Wireshark packet capture when CPE/UE has a successful session towards a data network. These information elements are standardized 

![image](https://github.com/nephio-project/api/assets/74044881/728852b6-81f9-4639-8349-67110b3f5a1f)